### PR TITLE
feat: validate min cpu & min memory are well under whats available

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
## Description:
This is a very best effort implementation. For docker we get a list of all running containers, and sum them; if one of the containers in the list dies we ignore them.

## Is this change user facing?
YES/NO

### CODE REVIEW

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go [hunk 1]
```diff
@@ -496,6 +500,14 @@ func (backend *DockerKurtosisBackend) DestroyDeprecatedCentralizedLogsResources(
 	return nil
 }
 
+func (backend *DockerKurtosisBackend) GetAvailableCPUAndMemory(ctx context.Context) (uint64, bool, uint64, bool, error) {
```
Comment (victorcolombo) [inline container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go]: can we create types so we don't have to deduce what these numbers means?

`type CpuMilli uint64`

Comment (h4ck3rk3y) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go]: Done @victorcolombo !

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go [hunk 2]
```diff
@@ -496,6 +501,14 @@ func (backend *DockerKurtosisBackend) DestroyDeprecatedCentralizedLogsResources(
 	return nil
 }
 
+func (backend *DockerKurtosisBackend) GetAvailableCPUAndMemory(ctx context.Context) (compute_resources.MemoryInMegaBytes, bool, compute_resources.CpuMilliCores, bool, error) {
```
Comment (victorcolombo) [inline container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go]: why we return `resourceInformationIsComplete` two times?

File: container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go [hunk 3]
```diff
@@ -1065,6 +1070,16 @@ func (manager *DockerManager) CopyFromContainer(ctx context.Context, containerId
 	return tarStreamReadCloser, nil
 }
 
+// GetAvailableCPUAndMemory returns free memory in megabytes, free cpu in millicores, information on whether cpu information is complete
```
Comment (victorcolombo) [inline container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go:1074]: use the same types you defined for the kurtosis backend?

**Review by victorcolombo — APPROVED** [review]

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go (hunk 2 above)
Comment (h4ck3rk3y) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go]: fixed :)

**Pull request merged**